Update setup steps for managed GCP identity (#3424)
astronomer/docs: astro/authorize-deployments-to-your-cloud.md

<TabItem value="gcp">

### Attach a service account to your Deployment

You can attach a custom GCP service account to your Deployment to grant the Deployment all of the service account's permissions.

> Using service accounts provides the greatest amount of flexibility for authorizing Deployments to your cloud. For example, you can use existing service accounts on new Deployments, or your can attach a single service account to multiple Deployments that should all have the same level of access to your cloud.

Reviewer: ```suggestion
Using service accounts provides the greatest amount of flexibility for authorizing Deployments to your cloud. For example, you can use existing service accounts on new Deployments, or your can attach a single service account to multiple Deployments that all have the same level of access to your cloud.
```

> 1. [Create a service account](https://cloud.google.com/iam/docs/service-accounts-create) in the GCP project that you want your Deployment to access. Grant the service account any permissions that the Deployment will need in your GCP project. Copy the service account ID to use later in this setup.

Reviewer 2: @jwitz we need to indicate that this option only exists on Hosted

Author: Updated via hotfix https://github.com/astronomer/docs/commit/7a9626b43b5bb82c30d656c1017d4db94168d157

2. In the Cloud UI, select your Deployment, then click **Details**. In the **Advanced** section, click **Edit**.

> 3. In the **Workload Identity** dropdown menu, select **Customer Managed Identity**

Reviewer: ```suggestion
3. In the **Workload Identity** menu, select **Customer Managed Identity**
```

<TabItem value="gcp">

### Attach a service account to your Deployment

> You can attach a custom GCP service account to your Deployment to grant the Deployment all of the permissions belonging to the service account. Astro uses [GCP service account impersonation](https://cloud.google.com/docs/authentication/use-service-account-impersonation) to allow your Deployment's workload identity to assume the configured service account. 

Reviewer 3: Astro uses [GCP service account impersonation](https://cloud.google.com/docs/authentication/use-service-account-impersonation) to allow your Deployment's workload identity to assume the configured service account. 

I think the above should be removed from this section as it describes impersonation.

Using service accounts provides the greatest amount of flexibility for authorizing Deployments to your cloud. For example, you can use existing service accounts on new Deployments, or your can attach a single service account to multiple Deployments that should all have the same level of access to your cloud.

1. [Create a service account](https://cloud.google.com/iam/docs/service-accounts-create) in the GCP project that you want your Deployment to access. Grant the service account any permissions that the Deployment will need in your GCP project. Copy the service account ID to use later in this setup.
2. In the Cloud UI, select your Deployment, then click **Details**. In the **Advanced** section, click **Edit**.

> 3. In the **Workload Identity** dropdown menu, select **ADD GCP Service Account**

Reviewer 3: ```suggestion
3. In the **Workload Identity** dropdown menu, select **Customer Managed Identity**
```

4. Enter your GCP service account ID when prompted, then copy and run the provided gcloud CLI command. 

> 5. Click **Save**. The service account is now selectable as a workload identity for the Deployment.

Reviewer 3: The user needs to **Update Deployment** to commit the change completely.

6. Complete one of the following options for your Deployment to access your cloud resources:

    - Create a **Google Cloud** connection type in Airflow and configure the following values:
      - **Connection Id**: Enter a name for the connection.

>       - **Impersonation Chain**: Enter the ID of the service account that your Deployment should impersonate.

Reviewer 3: not required as there is no impersonation happening with this option.

Reviewer 3: According to Mike's PR, the user should enter the project id of the target data service
https://github.com/astronomer/astro/pull/18523

@claassen was this required?

Reviewer 4: The projectID in the connection I created was just so that the hook knows what project to look at when connecting to the storage account, it is nothing to do with the authentication

>     Note that this implementation requires `apache-airflow-providers-google >= 10.8.0`. See [Add Python, OS-level packages, and Airflow providers](https://docs.astronomer.io/astro/cli/develop-project#add-python-os-level-packages-and-airflow-providers).

Reviewer 3: @melugoyal @claassen is this true?

Reviewer 4: From https://airflow.apache.org/docs/apache-airflow-providers-google/stable/changelog.html#id32 it looks like 10.8.0 introduced support for specifying the impersonation chain in the connection

Reviewer 3: this section is for customer managed identity. i don't believe there is a min version requirement for that to work.

>    - Specify the impersonation chain in code when you instantiate a Google Cloud operator. See [Airflow documentation](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/connections/gcp.html#direct-impersonation-of-a-service-account). Note that if you configure both a connection type and an operator, the operator-level configuration takes precedence.

Reviewer 3: not required for this option

    - To access resources in a secrets backend, run the following command to create an environment variable that grants access to the secrets backend:

    ```zsh
    astro deployment variable create --deployment-id <your-deployment-id> AIRFLOW__SECRETS__BACKEND_KWARGS={"connections_prefix": "airflow-connections", "variables_prefix": "airflow-variables", "project_id": "<your-secret-manager-project-id>", "impersonation_chain": "<your-gcp-service-account>"}
    ```

### Alternative setup: Authorize your Deployment through GCP service account impersonation

> If your organization has requirements over how service accounts are managed outside of your cloud, you can manually configure [GCP service account impersonation](https://cloud.google.com/docs/authentication/use-service-account-impersonation) to allow your Deployment's default workload identity to assume a service account in your GCP project.

Author: @melugoyal Does this sound right?

> If your organization has requirements over how service accounts are managed outside of your cloud, you can manually configure [GCP service account impersonation](https://cloud.google.com/docs/authentication/use-service-account-impersonation) to allow your Deployment's default workload identity to assume a service account in your GCP project.

Author: ```suggestion
If your organization has requirements over how service accounts are managed outside of your cloud, you can manually configure [GCP service account impersonation](https://cloud.google.com/docs/authentication/use-service-account-impersonation) to allow your Deployment's default workload identity to impersonate a service account in your GCP project.
```
